src/perception/ObjectDatabase.cpp

```diff
@@ -0,0 +1,85 @@
+#include <aikido/perception/ObjectDatabase.hpp>
+
+#include <dart/common/Console.hpp>
+#include <dart/common/LocalResourceRetriever.hpp>
+#include <yaml-cpp/exceptions.h>
+
+namespace aikido {
+namespace perception {
+
+//==============================================================================
+ObjectDatabase::ObjectDatabase(
+    const dart::common::ResourceRetrieverPtr& resourceRetriever,
+    dart::common::Uri configDataURI)
+{
+  // Read the JSON file into string
+  if (!resourceRetriever)
+  {
+    throw std::invalid_argument("ResourceRetrieverPtr given is null!");
+  }
+
+  const dart::common::ResourcePtr resource
+      = resourceRetriever->retrieve(configDataURI);
+  if (!resource)
+  {
+    throw std::runtime_error(
+        std::string("Failed opening URI - ") + configDataURI.toString());
+  }
+
+  // Put file in string
+  const std::size_t size = resource->getSize();
+  std::string content;
+  content.resize(size);
+  if (resource->read(&content.front(), size, 1) != 1)
+  {
+    throw std::runtime_error(
+        std::string("Failed reading  URI - ") + configDataURI.toString());
+  }
+
+  // Load from string
+  mObjData = YAML::Load(content);
+}
+
+//==============================================================================
```

Comment:
It seems that this function will either throw an error or succeed. I think rather than returning `bool` (which isn't very helpful if we're throwing errors anyway), we can just make the return type `void`.

Reply:
I agree with this point. I changed the return type to `void`, and removed all returns.